src/btree/row_srch.c

```diff
@@ -356,13 +370,28 @@ restart:	page = current->page;
 	}
 
 	/*
-	 * Binary search of the leaf page.  There are two versions (a default
-	 * loop and an application-specified collation loop), because moving
-	 * the collation test and error handling inside the loop costs about 5%.
+	 * Binary search of the leaf page. Three versions (record numbers, no
+	 * application-specified collation order and application-specified
+	 * collation order), because moving those tests and the error handling
+	 * inside the loop costs about 5%.
```

Comment:
Similarly, this comment should be updated if you take the "short key comparison" change.